### DISCUSSION TITLE
Content Sync on AEM cloud

### DISCUSSION THREAD

**Discussion by nrabbat** [Adobe-Consulting-Services/acs-aem-commons#3449] [Q&A] [answered]
Hi @YegorKozlov -
 
Great job on Content Sync.  My company is currently attempting to integrating it with our AEM Cloud setup. However, we ran into two things I wanted to share/ask about: 
 
First is with the AEM Cloud IP Allow Lists.  We noticed enabling them for on an AEM Cloud Instance source environment leads to a 403 error. Since Adobe doesn't offer a specific IP range for its instances that we can add to the Allow List, and we can't configure Content Sync to use a specific Egress IP on our side, it looks like we might need to disable IP Allow Listing to get the tool to function properly. You may want to include that in your documentation: It took us some time to realize that was the cause of the 403. 
 
Secondly, I attempted to deploy Content Sync on Publish in the Cloud.  There is no authentication on the Cloud for Publish instances (with on-prem AEM you can login as admin user), so I am assuming we would need to implement some kind of authentication ourselves to use Content Sync with Publish.  Can you confirm?  Am I missing something? 
 
Thanks again for the feature.  
 
-Nick

**Comment by YegorKozlov** [comment, marked as answer]
> First is with the AEM Cloud IP Allow Lists. We noticed enabling them for on an AEM Cloud Instance source environment leads to a 403 error. Since Adobe doesn't offer a specific IP range for its instances that we can add to the Allow List, and we can't configure Content Sync to use a specific Egress IP on our side, it looks like we might need to disable IP Allow Listing to get the tool to function properly. You may want to include that in your documentation: It took us some time to realize that was the cause of the 403.

Good catch. Content Sync  is not compatible with IP allow lists. If you a user wants to use Content Sync, IP allow lists must be disabled. I'll update the documentation.


> Secondly, I attempted to deploy Content Sync on Publish in the Cloud. There is no authentication on the Cloud for Publish instances (with on-prem AEM you can login as admin user), so I am assuming we would need to implement some kind of authentication ourselves to use Content Sync with Publish. Can you confirm? Am I missing something?

I don't recommend sync-ing publishers directly. A much better way is to run Content Sync on Author and run a workflow that would publish  each sync-ed page/asset. At least, this is how I designed it.
